fees/SmarDex/index.ts

```diff
@@ -1,52 +1,107 @@
-import { getDexChainFees } from "../../helpers/getUniSubgraphFees";
-import volumeAdapter from "../../dexs/SmarDex";
 import { CHAIN } from "../../helpers/chains";
-import type { Adapter } from "../../adapters/types";
+import type { Adapter, FetchResult } from "../../adapters/types";
+import { request, gql } from "graphql-request";
+import BigNumber from "bignumber.js";
 
-// Define fees for each chain
+import { getTimestampAtStartOfDayUTC } from "../../utils/date";
+
+const SMARDEX_SUBGRAPH_API_KEY = process.env.SMARDEX_SUBGRAPH_API_KEY || "";
+const SMARDEX_SUBGRAPH_GATEWAY = "https://subgraph.smardex.io/defillama";
+
+// Headers for GraphQL requests that require an API key
+const defaultHeaders = {
+  "x-api-key": SMARDEX_SUBGRAPH_API_KEY,
+};
+
+// Default fees for each chain
 const FEES = {
   [CHAIN.ETHEREUM]: { LP_FEES: 0.0005, POOL_FEES: 0.0002 },
   [CHAIN.BSC]: { LP_FEES: 0.0007, POOL_FEES: 0.0003 },
   [CHAIN.POLYGON]: { LP_FEES: 0.0007, POOL_FEES: 0.0003 },
   [CHAIN.ARBITRUM]: { LP_FEES: 0.0007, POOL_FEES: 0.0003 },
 } as { [chain: string]: { LP_FEES: number; POOL_FEES: number } };
 
-const adapter: Adapter = {
-  adapter: {},
+// SDEX contract creation timestamps for each chain
+const CHAIN_STARTS = {
+  [CHAIN.ETHEREUM]: 1678404995,
+  [CHAIN.BSC]: 1688978540,
+  [CHAIN.POLYGON]: 1682085480,
+  [CHAIN.ARBITRUM]: 1688976153,
+} as { [chain: string]: number };
+
+// Methodology descriptions
+const FEES_METHODOLOGY = `
+A minor fee is collected on each swap, functioning as trading fees.
+The fees are set at 0.07% on Ethereum and 0.1% on other chains.
+On other networks, the fees may vary between different pairs and chains.
+Refer to https://docs.smardex.io/overview/what-is-smardex/fees for detailed information.
+`;
+
+const methodology = {
+  UserFees: FEES_METHODOLOGY,
+  Fees: FEES_METHODOLOGY,
+  Revenue: `0.02% of each swap on Ethereum is collected for the staking pool (SDEX holders that staked).`,
+  ProtocolRevenue: `Protocol has no revenue.`,
+  SupplySideRevenue: `0.05% of each swap on Ethereum is collected for the liquidity providers.`,
+  HoldersRevenue: `0.02% of each swap on Ethereum is collected for the staking pool (SDEX holders that staked).`,
```

Comment:
Don't we need to adapt these lines for each chain ?

Reply:
I don't see the methodology by chain, I only see one on Defillama
IMO they use the methodology of the first adapter: ethereum

But it's a good point. If I add a methodology for each L2 which fees should I define ? The current fees or the default fees ?

Reply:
You can write something like this for revenues: eg: `Revenue: Fees are collected at each swap. Amount collected depends on pair traded. Default fees are 0.03%`

Reply:
Same idea for Supply/Holder ?

Reply:
yes ! If you can find a way to differentiate each chain